### PR TITLE
Add tagging permissions

### PR DESCRIPTION
**Pull request by martenvd**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## Explanation of the changes
My team is currently working on getting all our Azure DevOps permissions in code. This is one of the last permissions that we still only implement with Powershell rather than Terraform. I would love to have all our permissions in Terraform.

## What about the current behavior has changed?

Implementing the security permissions for the `tagging` feature in Azure DevOps.

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: azuredevops/internal/service/permissions/resource_tagging_permissions.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	return &schema.Resource{
+		Create: resourceTaggingPermissionsCreateOrUpdate,
+		Read:   resourceTaggingPermissionsRead,
+		Update: resourceTaggingPermissionsCreateOrUpdate,
+		Delete: resourceTaggingPermissionsDelete,
+		Schema: securityhelper.CreatePermissionResourceSchema(map[string]*schema.Schema{
+			"project_id": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.IsUUID,
+				Required:     true,
```
Comment (tmeckel) [inline azuredevops/internal/service/permissions/resource_tagging_permissions.go]: Can be `optional` to set global (organization wide) permissions for Tagging

File: azuredevops/internal/service/permissions/resource_tagging_permissions.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		return err
+	}
+	d.SetId("")
+	return nil
+}
+
+func createTaggingToken(d *schema.ResourceData, clients *client.AggregatedClient) (string, error) {
+	projectID, ok := d.GetOk("project_id")
+	if !ok {
+		return "ProjectID", nil
```
Comment (tmeckel) [inline azuredevops/internal/service/permissions/resource_tagging_permissions.go]: If `project_id` is not specified return an empty string `""` as token.

File: website/docs/r/tagging_permissions.html.markdown [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    Delete       = "allow"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `project_id` - (Required) The ID of the project to assign the permissions.
```
Comment (tmeckel) [inline website/docs/r/tagging_permissions.html.markdown]: `optional`: if not set, organization wide permissions for tagging are managed.

File: website/docs/r/tagging_permissions.html.markdown [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  Manages permissions for AzureDevOps Tagging
+---
+
+# azuredevops_tagging_permissions
+
+Manages permissions for tagging
+
+## Permission levels
+
+Permissions for tagging within Azure DevOps can be applied only on Project level.
```
Comment (tmeckel) [inline website/docs/r/tagging_permissions.html.markdown]: Can be applied on a specific project and globally on the organization

**Review by tmeckel — CHANGES_REQUESTED** [review]

**Comment by martenvd** [comment]
@tmeckel Thank you for the comments, I have resolved them 😃

File: azuredevops/internal/acceptancetests/resource_tagging_permissions_test.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+				Check: resource.ComposeTestCheckFunc(
+					testutils.CheckProjectExists(projectName),
+					resource.TestCheckResourceAttrSet(tfNodeRoot, "project_id"),
+					resource.TestCheckResourceAttrSet(tfNodeRoot, "principal"),
+					resource.TestCheckNoResourceAttr(tfNodeRoot, "path"),
+					resource.TestCheckResourceAttr(tfNodeRoot, "permissions.%", "4"),
+					resource.TestCheckResourceAttr(tfNodeRoot, "permissions.ViewSubscriptions", "deny"),
+					resource.TestCheckResourceAttr(tfNodeRoot, "permissions.EditSubscriptions", "notset"),
+					resource.TestCheckResourceAttr(tfNodeRoot, "permissions.DeleteSubscriptions", "deny"),
+					resource.TestCheckResourceAttr(tfNodeRoot, "permissions.PublishEvents", "deny"),
```
Comment (xuzhang3) [inline azuredevops/internal/acceptancetests/resource_tagging_permissions_test.go]: Test check does not map the HCl configures above.

**Review by xuzhang3 — CHANGES_REQUESTED** [review]

File: azuredevops/internal/acceptancetests/resource_tagging_permissions_test.go (hunk 5 above)
Comment (martenvd) [reply azuredevops/internal/acceptancetests/resource_tagging_permissions_test.go]: @xuzhang3 thank you for reviewing, forgot this copy paste error, but resolved it now.

**Comment by xuzhang3** [comment]
```log
=== RUN   TestAccTaggingPermissions_SetPermissions
=== PAUSE TestAccTaggingPermissions_SetPermissions
=== RUN   TestAccTaggingPermissions_UpdatePermissions
=== PAUSE TestAccTaggingPermissions_UpdatePermissions
=== CONT  TestAccTaggingPermissions_SetPermissions
=== CONT  TestAccTaggingPermissions_UpdatePermissions
--- PASS: TestAccTaggingPermissions_UpdatePermissions (108.03s)
--- PASS: TestAccTaggingPermissions_SetPermissions (127.20s)
PASS

```

**Comment by xuzhang3** [comment]
LGTM

**Comment by RodrigoGroener** [comment]
I've been waiting for the tagging permissions. Thank you for your contribution! When will be the next release of the provider including this feature?

**Pull request merged**